File: frontend/src/screens/LeaderboardScreen.js
import React, { useEffect, useState } from 'react';
import {
  View, Text, StyleSheet, FlatList, ActivityIndicator, TouchableOpacity,
} from 'react-native';
import { LinearGradient } from 'expo-linear-gradient';
import { getLeaderboardAPI } from '../services/api';

const MEDALS = ['🥇', '🥈', '🥉'];

export default function LeaderboardScreen() {
  const [leaders, setLeaders] = useState([]);
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState('');

  useEffect(() => {
    fetchLeaderboard();
  }, []);

  const fetchLeaderboard = async () => {
    setLoading(true);
    setError('');
    try {
      const res = await getLeaderboardAPI();
      setLeaders(res.data);
    } catch (err) {
      setError('Failed to load leaderboard. Check server connection.');
    } finally {
      setLoading(false);
    }
  };

  const renderItem = ({ item, index }) => {
    const winRate = item.gamesPlayed > 0
      ? Math.round((item.wins / item.gamesPlayed) * 100)
      : 0;

    return (
      <View style={[styles.row, index === 0 && styles.rowFirst]}>
        <Text style={styles.rank}>
          {index < 3 ? MEDALS[index] : `#${index + 1}`}
        </Text>
        <View style={styles.avatarBox}>
          <Text style={styles.avatarText}>
            {item.name.charAt(0).toUpperCase()}
          </Text>
        </View>
        <View style={styles.playerInfo}>
          <Text style={styles.playerName}>{item.name}</Text>
          <Text style={styles.playerStats}>
            {item.wins}W · {item.losses}L · {item.gamesPlayed} games
          </Text>
        </View>
        <View style={styles.winBox}>
          <Text style={styles.winNum}>{winRate}%</Text>
          <Text style={styles.winLbl}>Win Rate</Text>
        </View>
      </View>
    );
  };

  return (
    <LinearGradient colors={['#0d0520', '#1a0a2e']} style={styles.gradient}>
      <View style={styles.container}>
        <Text style={styles.title}>🏆 Leaderboard</Text>
        <Text style={styles.subtitle}>Top players by wins</Text>

        {loading ? (
          <ActivityIndicator size="large" color="#c084fc" style={{ marginTop: 40 }} />
        ) : error ? (
          <View style={styles.errorBox}>
            <Text style={styles.errorText}>{error}</Text>
            <TouchableOpacity style={styles.retryBtn} onPress={fetchLeaderboard}>
              <Text style={styles.retryText}>Retry</Text>
            </TouchableOpacity>
          </View>
        ) : leaders.length === 0 ? (
          <Text style={styles.emptyText}>No players yet. Be the first to play!</Text>
        ) : (
          <FlatList
            data={leaders}
            keyExtractor={(item) => item._id}
            renderItem={renderItem}
            showsVerticalScrollIndicator={false}
            contentContainerStyle={{ paddingBottom: 40 }}
          />
        )}
      </View>
    </LinearGradient>
  );
}

const styles = StyleSheet.create({
  gradient: { flex: 1 },
  container: { flex: 1, padding: 24, paddingTop: 60 },
  title: { fontSize: 26, fontWeight: '900', color: '#c084fc', marginBottom: 4 },
  subtitle: { color: '#6b7280', fontSize: 13, marginBottom: 24 },
  row: {
    flexDirection: 'row', alignItems: 'center', backgroundColor: '#1e1035',
    borderRadius: 14, padding: 14, marginBottom: 10,
    borderWidth: 1, borderColor: '#374151',
  },
  rowFirst: { borderColor: '#fbbf24', borderWidth: 2 },
  rank: { fontSize: 20, width: 36, textAlign: 'center' },
  avatarBox: {
    width: 40, height: 40, borderRadius: 20, backgroundColor: '#4c1d95',
    alignItems: 'center', justifyContent: 'center', marginHorizontal: 10,
  },
  avatarText: { color: '#fff', fontWeight: '700', fontSize: 16 },
  playerInfo: { flex: 1 },
  playerName: { color: '#f3f4f6', fontWeight: '700', fontSize: 15 },
  playerStats: { color: '#9ca3af', fontSize: 12, marginTop: 2 },
  winBox: { alignItems: 'center' },
  winNum: { color: '#4ade80', fontWeight: '800', fontSize: 18 },
  winLbl: { color: '#6b7280', fontSize: 11 },
  errorBox: { alignItems: 'center', marginTop: 40 },
  errorText: { color: '#f87171', fontSize: 14, textAlign: 'center', marginBottom: 16 },
  retryBtn: {
    backgroundColor: '#7c3aed', paddingHorizontal: 24, paddingVertical: 10, borderRadius: 10,
  },
  retryText: { color: '#fff', fontWeight: '700' },
  emptyText: { color: '#9ca3af', fontSize: 14, marginTop: 40, textAlign: 'center' },
});

File: flutter_app/lib/screens/leaderboard_screen.dart
import 'package:flutter/material.dart';
import '../theme.dart';
import '../services/api_service.dart';

class LeaderboardScreen extends StatefulWidget {
  const LeaderboardScreen({super.key});

  @override
  State<LeaderboardScreen> createState() => _LeaderboardScreenState();
}

class _LeaderboardScreenState extends State<LeaderboardScreen> {
  List<dynamic> _leaders = [];
  bool _loading = true;
  String? _error;

  final List<String> _medals = ['🥇', '🥈', '🥉'];

  @override
  void initState() {
    super.initState();
    _fetchLeaderboard();
  }

  Future<void> _fetchLeaderboard() async {
    setState(() {
      _loading = true;
      _error = null;
    });
    try {
      final res = await ApiService.getLeaderboard();
      setState(() => _leaders = res);
    } catch (e) {
      setState(() => _error = 'Failed to load leaderboard. Check server connection.');
    } finally {
      setState(() => _loading = false);
    }
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Container(
        width: double.infinity,
        decoration: const BoxDecoration(
          gradient: LinearGradient(
            colors: [AppColors.bg0, AppColors.bg1],
            begin: Alignment.topCenter,
            end: Alignment.bottomCenter,
          ),
        ),
        child: SafeArea(
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.start,
            children: [
              Padding(
                padding: const EdgeInsets.all(24),
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: const [
                    Text('🏆 Leaderboard', style: TextStyle(fontSize: 26, fontWeight: FontWeight.w900, color: AppColors.purpleLight)),
                    Text('Top players by wins', style: TextStyle(color: AppColors.textDim, fontSize: 13)),
                  ],
                ),
              ),
              Expanded(
                child: _buildContent(),
              ),
            ],
          ),
        ),
      ),
    );
  }

  Widget _buildContent() {
    if (_loading) {
      return const Center(child: CircularProgressIndicator(color: AppColors.purpleLight));
    }
    if (_error != null) {
      return Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: [
            Text(_error!, style: const TextStyle(color: AppColors.red)),
            const SizedBox(height: 16),
            ElevatedButton(
              onPressed: _fetchLeaderboard,
              style: ElevatedButton.styleFrom(backgroundColor: AppColors.purple),
              child: const Text('Retry'),
            ),
          ],
        ),
      );
    }
    if (_leaders.isEmpty) {
      return const Center(child: Text('No players yet. Be the first to play!', style: TextStyle(color: AppColors.textMuted)));
    }

    return ListView.builder(
      padding: const EdgeInsets.symmetric(horizontal: 24, vertical: 8),
      itemCount: _leaders.length,
      itemBuilder: (context, index) {
        final item = _leaders[index];
        final winRate = item['gamesPlayed'] > 0 ? (item['wins'] / item['gamesPlayed'] * 100).round() : 0;
        
        return Container(
          margin: const EdgeInsets.bottom(10),
          padding: const EdgeInsets.all(14),
          decoration: BoxDecoration(
            color: AppColors.card,
            borderRadius: BorderRadius.circular(14),
            border: Border.all(color: index == 0 ? AppColors.yellow : AppColors.borderAlt, width: index == 0 ? 2 : 1),
          ),
          child: Row(
            children: [
              Text(
                index < 3 ? _medals[index] : '#${index + 1}',
                style: const TextStyle(fontSize: 20),
              ),
              const SizedBox(width: 10),
              Container(
                width: 40,
                height: 40,
                decoration: const BoxDecoration(color: AppColors.purpleDark, shape: BoxShape.circle),
                alignment: Alignment.center,
                child: Text(
                  item['name']?.substring(0, 1).toUpperCase() ?? '',
                  style: const TextStyle(color: Colors.white, fontWeight: FontWeight.w700),
                ),
              ),
              const SizedBox(width: 10),
              Expanded(
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    Text(item['name'] ?? '', style: const TextStyle(color: AppColors.textPrimary, fontWeight: FontWeight.w700, fontSize: 15)),
                    Text('${item['wins']}W · ${item['losses']}L · ${item['gamesPlayed']} games', style: const TextStyle(color: AppColors.textMuted, fontSize: 12)),
                  ],
                ),
              ),
              Column(
                children: [
                  Text('$winRate%', style: const TextStyle(color: AppColors.green, fontWeight: FontWeight.w800, fontSize: 18)),
                  const Text('Win Rate', style: TextStyle(color: AppColors.textDim, fontSize: 11)),
                ],
              ),
            ],
          ),
        );
      },
    );
  }
}
